config/config.go

```diff
@@ -0,0 +1,49 @@
+// MIT License. Copyright (c) 2020 CQFN
+// https://github.com/cqfn/degitx/blob/master/LICENSE
+
+// Package config contains config parser code that is shared between front- and back-end parts.
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/urfave/cli/v2"
+)
+
+type Config interface {
+	FromFile(fileName string) error
+}
+
+type errConfigNotFound struct{ paths []string }
+
+func (e *errConfigNotFound) Error() string {
+	return fmt.Sprintf("configuration file not found in {%s}",
+		strings.Join(e.paths, ":"))
+}
+
+func ParseConfig(ctx *cli.Context, config Config, preDeclaredPaths ...string) error {
```

Comment:
@stepanvalyavskiy I don't think it's a good practice to make a dependency in lower level to upper level: now configuration depends on CLI library, so CLI depends on `config` and `config` depends on CLI deps. Also, I'd follow a single style for config parsing and make `Config` method receiver: `func (c *Config) FromFiles(files ...string) error`